contracts/OXO.sol

```diff
@@ -0,0 +1,145 @@
+pragma solidity ^0.4.15;
+
+contract OXO {
+ 
+    event BidEvent(address addr, uint ammount);
+    event MarkEvent(address addr, uint x, uint y);
+    event ErrorEvent(string message);
+    event WinnerEvent(uint8 winner);
+
+    address player1;
+    address player2;
+    uint player1Bid;
+    uint player2Bid;
+    uint8[3][] grid;
+    uint8 lastMarkByPlayer = 0;
+    uint lastMarkTimestamp = 0;
+    
+    function OXO() public {
+        grid = new uint8[3][](3);
+    }
+
+    function bidAndPlay() public payable returns (bool) {
+        if (player1==0){
+            if (player2!=0){
+                require(msg.value == player2Bid);
+            }
+            player1 = msg.sender;
+            player1Bid = msg.value;
+        } else if (player2==0){ 
+            require(msg.value == player1Bid);
+            player2 = msg.sender;
+            player2Bid = msg.value;
+        } else {
+            revert();
+        }
+        BidEvent(msg.sender, msg.value);
+        return true;
+    }
+    
+    // Money can be withdrawn if:
+    // 1) Game is not started yet (nobody put a first mark)
+    // 2) // TODO One of the players didn't put mark for an hour. In that case all bank can be withdrown by other player
+    function withdrawMoney() public returns(bool) {
+        if(lastMarkByPlayer==0){
+            if(msg.sender==player1){
+                 player1.transfer(player1Bid);
+                 player1 = 0;
+            } else if(msg.sender==player2){
+                 player2.transfer(player2Bid);
+                 player2 = 0;
+            }
+        }
+    }
+    
+    function mark(uint x, uint y) public returns (bool) {
+        require (player1Bid>0 && player2Bid>0); 
+        
+        // players can't put 2 marks in a row
+        if (msg.sender==player1){               
+            require(lastMarkByPlayer!=1);
+        } else if (msg.sender==player2){
+            require(lastMarkByPlayer!=2);
+        } else {
+            ErrorEvent("Somebody else tries to intervine");
+        }
+        
+        // out of grid check
+        require(x<3 && y<3);
+       
+        // already taken
+        require (grid[x][y]==0);
```

Comment:
sometimes you have spaces between signs and sometimes don't. Why?

Reply:
I am used to auto-fomat in IDE's. Otherwise I don't care about spaces. Is there any auto-format in sublime or remix?

Reply:
I use WebStorm or Atom, they are configured as you like